lib/puppet/type/kernel_parameter.rb

```diff
@@ -37,4 +39,25 @@
   autorequire(:file) do
     self[:target]
   end
+
+  # title_patterns method for mapping titles to namevars for supporting
+  # composite namevars.
+  # https://github.com/puppetlabs/puppetlabs-java_ks/blob/5bc34745a6f86e0c4af495e0ad3559c82d57873a/lib/puppet/type/java_ks.rb#L209
+  def self.title_patterns
+    [
+      [
+        %r{^([^:]+)$},
+        [
+          [:name],
+        ],
+      ],
+      [
+        %r{^(.*):(.*)$},
```

Comment:
For consistency does it make sense to match the other regexp that only match a name?  Also does an empty bootmode make sense?

```suggestion
        %r{^([^:]+):(.+)$},
```

If a bootmode should not have a `:` too, better math it too:

```suggestion
        %r{^([^:]+):([^:]+)$},
```